src/category_theory/monoidal/subcategory.lean

```diff
@@ -0,0 +1,159 @@
+/-
+Copyright (c) 2022 Antoine Labelle. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Labelle
+-/
+import category_theory.monoidal.braided
+import category_theory.concrete_category.basic
+
+/-!
+# Full monoidal subcategories
+
+Given a monidal category `C` and a monoidal predicate on `C`, that is a function `P : C → Prop`
+closed under `𝟙_` and `⊗`, we can put a monoidal structure on `{X : C // P X}` (the category
+structure is defined in `category_theory.full_subcategory`).
+
+When `C` is also braided/symmetric, the full monoidal subcategory also inherits the
+braided/symmetric structure.
+-/
+
+universes u v
+
+namespace category_theory
+
+namespace monoidal_category
+
+open iso
+
+variables (C : Type u) [category.{v} C] [monoidal_category C]
+
+/--
+A property of objects of a monoidal category `C` which is closed under `𝟙_` and `⊗`.
+-/
+@[nolint has_inhabited_instance]
+structure monoidal_predicate :=
+  (P : C → Prop)
+  (h_id : P (𝟙_ C))
+  (h_tensor : ∀ {X Y}, P X → P Y → P (X ⊗ Y))
```

Comment:
No indent needed.

Reply:
I suspect we may be happier later if we parametrize by `P` rather than have it be a bundled field. Not sure though. What do you think?

Reply:
Hum. Now that I think of it, if we parametrize by `P` and make this a typeclass, is the type synonym `full_monoidal_subcategory` really needed? I think we could then just do
```
instance (P : C → Prop) [monoidal_predicate P] : monoidal_category {X : C // P X}
```

Reply:
I think that might be a better solution especially since otherwise we're going to have the same issue with `monoidal_closed_category`, `rigid_category`, etc. and it doesn't seem very efficient to define type synonyms for `{X : C // P X}` for all these types of subcategories.

Reply:
Looks great!